### ISSUE TITLE
Redundant namespace for imported types

### ISSUE THREAD

**Issue by Nevon** [mtth/avsc#281]
Given the following AVDL files:

import.avdl:
```
@namespace("com.org.domain.fixtures")
protocol ImportProto {
  import idl "simple.avdl";

  record Import {
    com.org.domain.fixtures.Simple simple;
  }
}
```

simple.avdl
```
@namespace("com.org.domain.fixtures")
protocol SimpleProto {
  record Simple {
    string foo;
  }
}
```

The resulting AVSC file looks like this:

```
{
    "namespace": "com.org.domain.fixtures",
    "type": "record",
    "name": "Import",
    "fields": [
    {
        "type": {
        "type": "record",
        "name": "Simple",
        "fields": [
            {
            "type": "string",
            "name": "foo"
            }
        ],
        "namespace": "com.org.domain.fixtures"
        },
        "name": "simple"
    }
    ]
}
```

Note how `Simple` has the same namespace as `ImportProto`, yet still has a namespace field. Compare this to the Java equivalent using `avro-tools`:

```
{
    "type": "record",
    "name": "Import",
    "namespace": "com.org.domain.fixtures",
    "fields": [
    {
        "name": "simple",
        "type": {
        "type": "record",
        "name": "Simple",
        "fields": [
            {
            "name": "foo",
            "type": "string"
            }
        ]
        }
    }
    ]
}
```

**Comment by mtth** [comment]
The namespace is conservatively added to all imports to ensure that unqualified type names are correct. The schemas should be equivalent however; is this causing issues on your side?

Looking quickly at the code, I think it should work to omit [setting an imported type's namespace](https://github.com/mtth/avsc/blob/e1e1a56426c86f2f5d87e82c75369b265c79a4f5/lib/specs.js#L137) when it is equal to the main one. I think this would address the redundancy in the example you mention and would be happy to review a PR for this.

**Comment by Nevon** [comment]
The generated schemas are valid, so it's not causing any severe problems for me. The reason I noticed was that [when testing `kafkajs/confluent-schema-registry`](https://github.com/kafkajs/confluent-schema-registry/pull/43/files#diff-8ab5992534068c84a491ec65368ee8bc) we convert avdl files to avsc using `avro-tools` and then compare them to the ones we generate with `avsc` to see that they are equivalent. I recently made a change to enable imports to work, and noticed that the tests would fail when importing types from the same namespace, as described in this issue.

> Looking quickly at the code, I think it should work to omit setting an imported type's namespace when it is equal to the main one.

That was my conclusion as well from briefly looking through the code yesterday. If I have the time, I will try to create a PR for this.
